app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -184,161 +181,217 @@ public boolean contactExists(final String name) throws IOException {
         return getContactWithName(name) != null;
     }
 
+
     /**
      * {@inheritDoc}
      */
     @Override
-    public List<Profile> getProfiles() throws IOException {
-        final GsonBuilder gsonBuilder = new GsonBuilder()
-                .registerTypeHierarchyAdapter(IKey.class, new GsonKeyAdapter());
-        final Gson gsonParser = gsonBuilder.create();
-
-        final Reader reader = this.fileSystem.getReader(this.profilesPath);
-        final List<Profile> profile = gsonParser.fromJson(reader, TYPE_PROFILE_LIST);
-        reader.close();
-
-        return profile;
+    public Profile getProfile() throws IOException {
+        return getDatabase().getProfile();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void addProfile(final Profile profile) throws IOException {
-        // Get the list of profiles and add the new profile
-        final List<Profile> profiles = this.getProfiles();
-        profiles.add(profile);
-
-        final GsonBuilder gsonBuilder = new GsonBuilder()
-                .registerTypeHierarchyAdapter(IKey.class, new GsonKeyAdapter());
-        final Gson gsonParser = gsonBuilder.create();
+    public void updateProfile(final Profile newProfile) throws IOException {
+        getDatabase().setProfile(newProfile);
+        updateDatabase();
 
-        // Update the database
-        final Writer writer = this.fileSystem.getWriter(this.profilesPath);
-        gsonParser.toJson(profiles, writer);
-        writer.close();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void deleteProfile(final Profile profile) throws IllegalArgumentException, IOException {
-        // Get the list of contacts
-        final List<Profile> profiles = this.getProfiles();
-        final int lengthBefore = profiles.size();
-        profiles.remove(profile);
+    public void updateDatabase() throws IOException {
+        try {
+            final GsonBuilder gsonBuilder = new GsonBuilder().registerTypeHierarchyAdapter(IKey.class, new GsonKeyAdapter());
+            final Gson gsonParser = gsonBuilder.create();
 
-        // Throw if the contact to remove is not found
-        if (profiles.size() == lengthBefore) {
-            throw new IllegalArgumentException(CONTACT_NOT_FOUND);
+            final String databaseJson = gsonParser.toJson(getDatabase());
+            final String encryptedDatabase = EncryptedSaver.encryptOrDecryptWithPassword(databaseJson, getEncryptionKey(), true);
+            final Writer writer = this.fileSystem.getWriter(databasePath);
```

Comment:
```
try (Writer writer = this.fileSystem.gerWriter(databasePath)) {
writer.write(encryptedDatabase);
}
```
If you do that, you don't have to use close(). It's safer to get accustomed to always using try() instead of close, because with try you cannot forget to write close(), resulting in a memory leak